internal/integration/rhp/v4/rhp4_test.go

```diff
@@ -0,0 +1,1383 @@
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func BenchmarkRead(b *testing.B) {
+	n, genesis := testutil.V2Network()
+	hostKey, renterKey := types.GeneratePrivateKey(), types.GeneratePrivateKey()
+
+	hn := testutil.NewHostNode(b, hostKey, n, genesis, zap.NewNop())
+	cm := hn.Chain
+	w := hn.Wallet
+
+	results := make(chan error, 1)
+	if _, err := hn.Volumes.AddVolume(context.Background(), filepath.Join(b.TempDir(), "test.dat"), uint64(b.N), results); err != nil {
+		b.Fatal(err)
+	} else if err := <-results; err != nil {
+		b.Fatal(err)
+	}
+
+	testutil.MineAndSync(b, hn, w.Address(), int(n.MaturityDelay+20))
+
+	transport := testRenterHostPair(b, hostKey, hn, zap.NewNop())
+
+	settings, err := rhp4.RPCSettings(context.Background(), transport)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	fundAndSign := &fundAndSign{w, renterKey}
+	renterAllowance, hostCollateral := types.Siacoins(100), types.Siacoins(200)
+	formResult, err := rhp4.RPCFormContract(context.Background(), transport, cm, fundAndSign, cm.TipState(), settings.Prices, hostKey.PublicKey(), settings.WalletAddress, proto4.RPCFormContractParams{
+		RenterPublicKey: renterKey.PublicKey(),
+		RenterAddress:   w.Address(),
+		Allowance:       renterAllowance,
+		Collateral:      hostCollateral,
+		ProofHeight:     cm.Tip().Height + 50,
+	})
+	if err != nil {
+		b.Fatal(err)
+	}
+	revision := formResult.Contract
+
+	// fund an account
+	cs := cm.TipState()
+	account := proto4.Account(renterKey.PublicKey())
+	accountFundAmount := types.Siacoins(25)
+	fundResult, err := rhp4.RPCFundAccounts(context.Background(), transport, cs, renterKey, revision, []proto4.AccountDeposit{
+		{Account: account, Amount: accountFundAmount},
+	})
+	if err != nil {
+		b.Fatal(err)
+	}
+	revision.Revision = fundResult.Revision
+
+	token := proto4.AccountToken{
+		Account:    account,
+		ValidUntil: time.Now().Add(time.Hour),
+	}
+	token.Signature = renterKey.SignHash(token.SigHash())
+
+	var sectors [][proto4.SectorSize]byte
+	roots := make([]types.Hash256, 0, b.N)
+	for i := 0; i < b.N; i++ {
+		var sector [proto4.SectorSize]byte
+		frand.Read(sector[:256])
+		sectors = append(sectors, sector)
+
+		// store the sector
+		writeResult, err := rhp4.RPCWriteSector(context.Background(), transport, settings.Prices, token, bytes.NewReader(sectors[i][:]), proto4.SectorSize)
+		if err != nil {
+			b.Fatal(err)
+		}
+		roots = append(roots, writeResult.Root)
+	}
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	b.SetBytes(proto4.SectorSize)
+
+	buf := bytes.NewBuffer(make([]byte, 0, proto4.SectorSize))
+	for i := 0; i < b.N; i++ {
+		buf.Reset()
+		// store the sector
+		_, err = rhp4.RPCReadSector(context.Background(), transport, settings.Prices, token, buf, roots[i], 0, proto4.SectorSize)
+		if err != nil {
+			b.Fatal(err)
+		} else if !bytes.Equal(buf.Bytes(), sectors[i][:]) {
+			b.Fatal("data mismatch")
+		}
+	}
+}
+
+func BenchmarkContractUpload(b *testing.B) {
```

Comment:
`nit` makes sense to add whatever result you get here on your machine

Ran them on mine in case you want to just grab that one:
```
goos: darwin
goarch: arm64
pkg: go.sia.tech/hostd/internal/integration/rhp/v4
cpu: Apple M1 Max
=== RUN   BenchmarkContractUpload
BenchmarkContractUpload
BenchmarkContractUpload-10            26          42209474 ns/op          99.37 MB/s     5027475 B/op    12994 allocs/op
PASS
ok      go.sia.tech/hostd/internal/integration/rhp/v4   5.687s
```